### PR TITLE
feat: add NPM package documentation

### PR DESCRIPTION
**Pull request by kpanot**

### CODE REVIEW

File: packages/@o3r/dev-tools/README.md [hunk 1]
```diff
@@ -1,3 +1,283 @@
...
+
+### Usage
+
+```bash
+Usage: version-harmonize [options]
+
+Replace the dependencies version in a monorepos
+
+Options:
+  -m, --monorepo <package>          Path to the private package.json of the monorepo (default: "<process.cwd()>")
```
Comment (cpaulve-1A) [inline packages/@o3r/dev-tools/README.md:233]: Missing 's' to the verbs

File: packages/@o3r/dev-tools/README.md [hunk 2]
```diff
@@ -1,3 +1,283 @@
...
+  -v, --verbose                     Display debug logs
+  -a, --alignPeerDependencies       Enforce to align the version of the dependencies with the latest range
+  -h, --help                        display help for command
+```
+
+### Otter options
+
+The `version-harmonize` command is reading the content of the package.json file to get the following options:
+
+* ignore: List of dependency name patterns to ignore when updating the dependencies range.
```
Comment (cpaulve-1A) [inline packages/@o3r/dev-tools/README.md:244]: Missing 's' on the verbs

File: packages/@o3r/dev-tools/README.md [hunk 3]
```diff
@@ -1,3 +1,283 @@
...
+        "webpack"
+      ]
+    }
+  }
+}
+```
+
+## Generate Package Exports
+
+Edit the generated package.json file to add the exports of the packages based on defined sub-entries.
```
Comment (cpaulve-1A) [inline packages/@o3r/dev-tools/README.md:265]: Edits - unless this is an order

File: packages/@o3r/dev-tools/README.md [hunk 4]
```diff
@@ -1,3 +1,283 @@
...
+
+### Usage
+
+```bash
+Usage: generate-package-exports [options]
+
+Update package.json exports
+
+Options:
+  --cwd <path>               Path to the root of the project (default: "<process.cwd()>")
```
Comment (cpaulve-1A) [inline packages/@o3r/dev-tools/README.md:276]: missing s on verbs

File: packages/@o3r/dynamic-content/README.md [hunk 5]
```diff
@@ -1,3 +1,114 @@
...
+
+```shell
+ng add @o3r/dynamic-content
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
+
+## Description
+
+In order to get your content from a different location from where your application is hosted, you may use the ``DynamicContentModule`` from ``@o3r/dynamic-content``. To include the module, you should just:
```
Comment (cpaulve-1A) [inline packages/@o3r/dynamic-content/README.md:17]: from a different location than ....

File: packages/@o3r/forms/README.md [hunk 6]
```diff
@@ -1,3 +1,94 @@
...
+
 This module provides utilities to enhance angular form (asynchronous decorator, additional validator, error store...).
+
+## How to install
+
+```shell
+ng add @o3r/forms
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:13]: requires

File: packages/@o3r/forms/README.md [hunk 7]
```diff
@@ -1,3 +1,94 @@
...
+
+## Container/presenter and form creation
+
+Container/presenter architecture was put in place to ensure best reusability/sharing
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:17]: A container/presenter

File: packages/@o3r/forms/README.md [hunk 8]
```diff
@@ -1,3 +1,94 @@
...
+
+* **form created in presenter** - it's the presenter which decides how the data is displayed
+* **container** needs only the value and errors propagated from the presenter
+* **container** can set the default value
+
+### How the container and presenter will communicate in forms context  
+
+* **presenter** implements [ControlValueAccessor](https://angular.io/api/forms/ControlValueAccessor) and [Validator](https://angular.io/api/forms/Validator) (or [AsyncValidator](https://angular.io/api/forms/AsyncValidator)) interfaces
+  * **propagate** the value, the form status and the errors
+* **container** apply [FormControlDirective](https://angular.io/api/forms/FormControlDirective) on the presenter html tag
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:29]: applies

File: packages/@o3r/forms/README.md [hunk 9]
```diff
@@ -1,3 +1,94 @@
...
+  * **container** set the default value using **formControl** directive
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:30]: sets

File: packages/@o3r/forms/README.md [hunk 10]
```diff
@@ -1,3 +1,94 @@
...
+  * _getFlatControlErrors_ function is available in **@o3r/forms** to help with the creation of error object model  
+* **custom** validators created at container level
+  * localization of the error messages associated done at container level
+  * custom validators are passed as an input to the presenter
+  * the error returned by the validator has to be compliant with the form error store model
+
+### Display errors on a messages panel
+
+* a dedicated _FormErrorStore_ is available on **@o3r/forms**
+  * allow the display of errors anywhere on the page
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:54]: allows

File: packages/@o3r/forms/README.md [hunk 11]
```diff
@@ -1,3 +1,94 @@
...
+  * the error object model contains the translation key and params  
+See [FORM_VALIDATION](https://github.com/AmadeusITGroup/otter/tree/main/docs/forms/FORM_VALIDATION.md) and [FORM_ERRORS](https://github.com/AmadeusITGroup/otter/tree/main/docs/forms/FORM_ERRORS.md)
+
+## Form submit
+
+### You want to submit the form
+
+* submit triggered by the submit button **in the presenter** and an event is emitted
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:62]: The submit is triggered

File: packages/@o3r/forms/README.md [hunk 12]
```diff
@@ -1,3 +1,94 @@
...
+* **container** capture the event and execute the submit form logic
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:63]: captures  / executes

File: packages/@o3r/forms/README.md [hunk 13]
```diff
@@ -1,3 +1,94 @@
...
+
+### What happens when you have multiple forms and you want to submit?
+
+The answer for this question is that we should avoid having multiple _form_ tags in the same page, as much as possible, because it adds a lot of complexity. We should try to have only one _form_ tag that encapsulates everything and one submit action.
+  
+If the case of multiple forms it's really needed, then we found the following solution:
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:69]: If multiple forms are really needed

File: packages/@o3r/forms/README.md [hunk 14]
```diff
@@ -1,3 +1,94 @@
...
+
+* submit button hidden on the presenters
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:71]: the submit button is hidden

File: packages/@o3r/forms/README.md [hunk 15]
```diff
@@ -1,3 +1,94 @@
...
+* the **submit** is **triggered from the page**
+* an **observable** to trigger the submit is passed as **input** to the containers;
+* _AsyncInput_ decorator is provided in **@o3r/forms** to be applied on the observable input to ensure performance
+* submit form logic is executed on the containers
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:75]: The submit form logic is

File: packages/@o3r/forms/README.md [hunk 16]
```diff
@@ -1,3 +1,94 @@
...
+* containers emit events when the submit is done
+* the page (parent) capture the events and continue it's logic
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:77]: captures / continues / its

File: packages/@o3r/forms/README.md [hunk 17]
```diff
@@ -1,3 +1,94 @@
...
+
+This can be applied also, with only one form on the page, when you don't want a submit button in the presenter.
+
+### What happens when you submit from page and the form was not touched
+
+At the first display of the form, the inline errors (if the form is invalid) are not displayed, because the form element is **not touched** and **dirty**
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:83]: no coma between because and the clause it refers to

File: packages/@o3r/forms/README.md [hunk 18]
```diff
@@ -1,3 +1,94 @@
...
+In the case you want to show the inline errors after the submit, you have to:
+
+* register a function in the container to _mark touched and dirty_ the form
+* the function is passed via an _@Output_ from the presenter and has to be called before executing the submit logic
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:87]: Pass the function ... and call it before...

File: packages/@o3r/forms/README.md [hunk 19]
```diff
@@ -1,3 +1,94 @@
...
+* _markAllControlsDirtyAndTouched_ helper is available in **@o3r/forms** to mark interactions on given form
```
Comment (cpaulve-1A) [inline packages/@o3r/forms/README.md:88]: use ... (as your previous sentence end with "you have to")

File: packages/@o3r/logger/README.md [hunk 20]
```diff
@@ -1,3 +1,97 @@
...
+
 This module provides a bridge to target different logger (logrocket, fullstory, smartlook) with a common interface.
+
+## How to install
+
+```shell
+ng add @o3r/logger
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/logger/README.md:13]: requires

File: packages/@o3r/mobile/README.md [hunk 21]
```diff
@@ -1,3 +1,13 @@
...
+
 This package contains tooling useful for mobile applications.
+
+## How to install
+
+```shell
+ng add @o3r/mobile
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/mobile/README.md:13]: requires

File: packages/@o3r/routing/README.md [hunk 22]
```diff
@@ -1,4 +1,18 @@
...
+This module helps you to configure an Otter application's routing
 (POST parameter management, preloading strategy, guard recommendation).
+
+## How to install
+
+```shell
+ng add @o3r/routing
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/routing/README.md:14]: requires

File: packages/@o3r/rules-engine/README.md [hunk 23]
```diff
@@ -1,4 +1,42 @@
...
 This module provides a rule engine that can be executed on your Otter application to customize your application (translations,
 placeholders and configs) based on a json file generated by your CMS
+
+## How to install
+
+```shell
+ng add @o3r/rules-engine
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/rules-engine/README.md:14]: requires

File: packages/@o3r/rules-engine/README.md [hunk 24]
```diff
@@ -1,4 +1,42 @@
...
+
+## How to use
+
+To facilitate the understanding of the usuage of the rule engines, the documentation has been split in several sessions:
```
Comment (cpaulve-1A) [inline packages/@o3r/rules-engine/README.md:18]: usage => typo  split into several sections

File: packages/@o3r/rules-engine/README.md [hunk 25]
```diff
@@ -1,4 +1,42 @@
...
+- [basic-rule](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/basic-rule.md)
+- [complex-fact](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/complex-fact.md)
+- [custom-operator](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/custom-operator.md)
+- [nested-conditions](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/nested-conditions.md)
+- [readme](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/readme.md)
+- [runtime-facts](https://github.com/AmadeusITGroup/otter/tree/main/docs/examples/runtime-facts.md)
+
+## Details
+
+More details regarding the way the rules engine is working can be find in the [documentation](https://github.com/AmadeusITGroup/otter/tree/main/docs/rules-engine/).
```
Comment (cpaulve-1A) [inline packages/@o3r/rules-engine/README.md:42]: can be found

File: packages/@o3r/storybook/README.md [hunk 26]
```diff
@@ -1,3 +1,112 @@
...
+> See [Story Helpers section](#stories-helpers) for more explanations.
+
+## Otter AddOns
+
+To facilitate the customization of an Otter component, 2 Storybook Addons are provided.
+
+### Global Theming tab
+
+- ***(1)***: To facilitate the administration of global theming of a component, a tab has been added to the default storybooks tabs.
+- ***(2)***: Each global theme variables are editable
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:56]: Each global theme variable is editable

File: packages/@o3r/storybook/README.md [hunk 27]
```diff
@@ -1,3 +1,112 @@
...
+- ***(3)***: 3 actions buttons are provided
+  - Theme switching button (visiable only in case of multi theme configured). See [Multi Theme section](#multi-theming).
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:58]: visible => typo

File: packages/@o3r/storybook/README.md [hunk 28]
```diff
@@ -1,3 +1,112 @@
...
+| **setLocalizationMetadata**: register the localization metadata    | *localization metadata* extracted by `@o3r/localization` from the application/library.                                                         | **extractLocalization**: retrieve full localization configuration for the component        | *translation*: current component translations map                                                                                                    |
+
+> Find documentation to write a story on [Storybook documentation](https://storybook.js.org/docs/react/writing-stories/introduction).
+
+## Additional features
+
+### Multi theming
+
+The Theme Addon is supporting multi theme and allow the user to switch from one theme to another.
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:87]: allows

File: packages/@o3r/storybook/README.md [hunk 29]
```diff
@@ -1,3 +1,112 @@
...
+
+```typescript
+import { getThemeVariables, setStyleMetadata } from '@o3r/storybook';
+
+// metadata extracted from my application
+import defaultStyleMetadata from '../dist/style.metadata.json';
+// metadata extracted from another theme of my application (will contains only the variable relative to this custom theme)
+import customStyleMetadata from '../dist/style.metadata.json';
+
+// registrer my application default style metadata
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:98]: Registers

File: packages/@o3r/storybook/README.md [hunk 30]
```diff
@@ -1,3 +1,112 @@
...
+setStyleMetadata({defaultStyleMetadata});
+
+
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
+  themes: {
+    // the function getThemeVariables() will extract, from metadata, only the CSS variable provided by the global theme, as key/value map
+    // set as default theme the map of the default theme css variable
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:106]: Sets

File: packages/@o3r/storybook/README.md [hunk 31]
```diff
@@ -1,3 +1,112 @@
...
+    default: getThemeVariables(),
+    // add additional theme by providing the map of css variable extracted from the custom theme
```
Comment (cpaulve-1A) [inline packages/@o3r/storybook/README.md:108]: Adds

File: packages/@o3r/stylelint-plugin/README.md [hunk 32]
```diff
@@ -1,3 +1,17 @@
...
+
 In-house stylint plugins to use in your own eslint configuration
+
+## How to install
+
+```shell
+ng add @o3r/styling-plugin
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/stylelint-plugin/README.md:13]: requires

File: packages/@o3r/stylelint-plugin/README.md [hunk 33]
```diff
@@ -1,3 +1,17 @@
...
+
+## Rules
+
+The list of rules are available in the [documentation](https://github.com/AmadeusITGroup/otter/tree/main/docs/linter/stylelint-plugin/rules.md).
```
Comment (cpaulve-1A) [inline packages/@o3r/stylelint-plugin/README.md:17]: is

File: packages/@o3r/styling/README.md [hunk 34]
```diff
@@ -1,3 +1,17 @@
...
+
 A styling framework to apply a theme on an Otter application at build time and runtime to support CMS customization
+
+## How to install
+
+```shell
+ng add @o3r/styling
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/styling/README.md:13]: requires

File: packages/@o3r/testing/README.md [hunk 35]
```diff
@@ -1,3 +1,17 @@
...
+
 The module provides testing (e2e, unit test) utilities to help you build your own E2E pipeline integrating visual testing
+
+## How to install
+
+```shell
+ng add @o3r/testing
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/testing/README.md:13]: requires

File: packages/@o3r/third-party/README.md [hunk 36]
```diff
@@ -1,3 +1,13 @@
...
+
 This module provides a bridge to communicate with third parties via an iFrame solution
+
+## How to install
+
+```shell
+ng add @o3r/third-party
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (cpaulve-1A) [inline packages/@o3r/third-party/README.md:13]: requires

File: packages/@o3r/localization/README.md [hunk 37]
```diff
@@ -1,3 +1,703 @@
...
+[ICU Message Format](http://userguide.icu-project.org/formatparse/messages) is a standardized syntax for dealing with the translation of user-visible strings into various languages that may have different requirements for the correct declension of words (e.g. according to number, gender, case) - or to simplify: pluralization.  
+Simple pluralization rules like *0, 1 or other* fits well for English but may not fit for many other languages (eastern Europe languages, asiatic languages) where pluralization rules are much more complex. If this does not fit your requirement we recommend to reformulate your text, so that you do not need to use pluralization. Example: instead of saying 'You have added 2 baggages' you may want to say 'Pieces of baggage: 2' which should be fine for most of languages no matter which number is considered to be plural.
+
+#### Integration with ngx-translate
+
+You need to configure *TranslateModule* so it uses *TranslateMessageFormatCompiler* as compiler. We will use *TranslateMessageFormatLazyCompiler* which is an otter extension of the base compiler. See *Lazy Compiler for ICU Translation support* above chapter for details.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:441]: remove "chapter"

File: packages/@o3r/localization/README.md [hunk 38]
```diff
@@ -1,3 +1,703 @@
...
+- *'Value of count messages' errors* if *countMessages* is greater than 1 (ex: 10 Errors)
+
+### How to localize a choice
+
+Sometimes you may want to display a different resource based on some property value which does not resolve to a number.
+
+```typescript
+// in localization bundle the key has to be defined
+"global.people": "{gender, select, male{He is} female{She is} other{They are}} {how}"
+// where 'gender' is the parameter used for choice and 'how' it's a parameter used only for display
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:496]: it's => is

File: packages/@o3r/localization/README.md [hunk 39]
```diff
@@ -1,3 +1,703 @@
...
+
+### Runtime: toggle translation on and off
+
+In order to be able to more easily identify which key corresponds to a given text, the ``LocalizationService`` exposes a function ``toggleShowKeys()`` that can be called in order to deactivate or reactivate the translation mechanism at **runtime**.  
+While deactivated, the ``translate`` **pipe** and **directive** will output the translation keys instead of their resolved values.
+
+> **Important**: this mechanism only applies to the pipe and directive exported by Otter's ``LocalizationModule``. The original ones from ``ngx-translate`` do not support it.
+
+First, this mechanism has to be activated via the ``LocalizationConfiguration`` that you can provide in your ``ApplicationModule``.  
+This is mainly for performances reason: the way it works is it adds a new subscription to every ``translate`` pipe and directive in order to know when translations are turned on or off.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:528]: performance reasons

File: packages/@o3r/localization/README.md [hunk 40]
```diff
@@ -1,3 +1,703 @@
...
+<!-- Component template -->
+<div>
+  <h3>Debug:</h3>
+  <button (click)="toggleTranslation()">Toggle translation</button>
+</div>
+````
+
+### Bootstrap: enable debug mode to display ``<key> - <translation>``
+
+You can enable debug mode by setting debugMode property of LocalizationConfiguration (by default the value of debugMode is false)
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:589]: setting **the** debugMode property of LocalizationConfiguration **to true**

File: packages/@o3r/localization/README.md [hunk 41]
```diff
@@ -1,3 +1,703 @@
...
+
+- [simpleHeader.airline] My Airline
+
+- [simpleHeader.motto] Let's shape the future of travel
+
+## How to properly create keys?
+
+It's always good to stick to some naming conventions.
+
+For components we prefix each key by component selector to ensure cross components keys uniqueness.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:623]: by **a** component selector to ensure **unique** cross components keys

File: packages/@o3r/localization/README.md [hunk 42]
```diff
@@ -1,3 +1,703 @@
...
+    "description": "airline name",
+    "defaultValue": "My Airline"
+  }
+}
+
+```
+
+## Intelligent fallback support
+
+The fallback hierarchy is been added to the localization service which explained in detail below. As the last option, the default fallback language picked as usual.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:644]: **has** been added
which **is** explained
As a last resort, the default fallback language is picked as usual.

File: packages/@o3r/localization/README.md [hunk 43]
```diff
@@ -1,3 +1,703 @@
...
+
+### Scenario 1: Fallback based on **fallbackLocalesMap**, **supportedLocales** language code
+
+Incase if **fallbackLocalesMap** provided and the targeted translation language is unavailable in supported locales list, The priority goes to **fallbackLocalesMap**, to see if targeted translation language can be mapped with the fallback map configured. The second priority goes to **supportedLocales**, to map targeted translation language to the first closest possible language, the locale can be different. If none match, it will fallback to the **default language**.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:648]: - If **fallbackLocalesMap** is provided ...
- locales list, the priority goes to **fallbackLocalesMap** to see if **the** targeted ... 
- to **supportedLocales** to map **the** targeted ... 
- closest possible language (the locale can be different).

File: packages/@o3r/localization/README.md [hunk 44]
```diff
@@ -1,3 +1,703 @@
...
+
+en-AU **fallbacks to** en-GB, as fallback locales mapping unavailable, first nearest language available in supported locales.
+
+fr-BE **fallbacks to** fr-FR, as fallback locales mapping unavailable, first nearest language available in supported locales.
+
+bn-BD **fallbacks to** ar-AR, as it is the default fallback.
+
+### Scenario 2: Fallback based on **supportedLocales** language code
+
+In case if **fallbackLocalesMap** is not provided and the targeted translation language is unavailable is supported locales list, The targeted translation language will be matched with supported locales **language** to find out the first nearest match, the **locale** can be different. If none match, it will fallback to the **default language**.
```
Comment (sdo-1A) [inline packages/@o3r/localization/README.md:686]: - If **fallbackLocalesMap** is not provided ... 
- unavailable **in** supported locales list, the targeted ... 
- to **find the** first **closest** match (the **locale** can be different).

File: packages/@o3r/logger/README.md [hunk 45]
```diff
@@ -1,3 +1,97 @@
...
+  ]
+})
+export class AppModule {}
+```
+
+## How to use
+
+The `LoggerService` is agnostic of the third-party services and offers an abstraction layer for the application.
+
+It exposes generic method that can be called directly regardless of the client used.
```
Comment (sdo-1A) [inline packages/@o3r/logger/README.md:73]: **a** generic method

File: packages/@o3r/rules-engine/README.md [hunk 46]
```diff
@@ -1,4 +1,42 @@
...
+- [create-custom-fact](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/create-custom-fact.md)
+- [debug-panel](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/debug-panel.md)
+- [debug](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/debug.md)
+- [dedicated-component-ruleset](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/dedicated-component-ruleset.md)
+- [performance-metrics](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/performance-metrics.md)
+- [placeholders](https://github.com/AmadeusITGroup/otter/tree/main/docs/how-to-use/placeholders.md)
+
+## Examples
+
+Several examples of the rule engine usage is available on the following links:
```
Comment (sdo-1A) [inline packages/@o3r/rules-engine/README.md:31]: **are** available

File: packages/@o3r/rules-engine/README.md (hunk 25 above)
Comment (sdo-1A) [reply packages/@o3r/rules-engine/README.md:42]: the rules engine works

File: packages/@o3r/schematics/README.md [hunk 47]
```diff
@@ -1,3 +1,17 @@
...
+
+This module provides basic utilities and generic rules (install, lint) reused in other package schematics.
+
+## How to install
+
+```shell
+ng add @o3r/schematics
+```
+
+> **Warning**: this module required [@o3r/core](https://www.npmjs.com/package/@o3r/core) to be installed.
```
Comment (sdo-1A) [inline packages/@o3r/schematics/README.md:13]: requires

File: packages/@o3r/storybook/README.md [hunk 48]
```diff
@@ -1,3 +1,112 @@
...
-This modules provides utilities to integrate the Storybook framework to your Otter project
+This package is an [Otter Framework Module](https://github.com/AmadeusITGroup/otter/tree/main/docs/core/MODULE.md).
+
+The Otter Framework provides several tools for Otter based projects.
+
+## Description
+
+The Otter Framework provides 3 sets of tools for Storybook integration:
+
+- **Generators**: To generate basic Story and setup the application to support Storybook
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:11]: set up

File: packages/@o3r/schematics/README.md [hunk 49]
```diff
@@ -1,3 +1,17 @@
...
+
+## Description
+
+This is a technical package used as dependency by [Otter modules](https://github.com/AmadeusITGroup/otter/tree/main/docs/core/MODULE.md) providing helpers to used in [Schematics](https://angular.io/guide/schematics) development.
```
Comment (sdo-1A) [inline packages/@o3r/schematics/README.md:17]: to **be** used

File: packages/@o3r/storybook/README.md [hunk 50]
```diff
@@ -1,3 +1,112 @@
...
+- **Stories helpers**: to create story controllers based on Otter framework
+- **AddOns**: to add feature to interact with Otter Customization
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:13]: features

File: packages/@o3r/storybook/README.md [hunk 51]
```diff
@@ -1,3 +1,112 @@
...
+
+When creating a new application/library (`ng new`), the Otter Framework can be added with the `ng add @o3r/core` command. When executing this command, the following question will be asked: *Add storybook setup?* (default value: `yes`).
+If `yes` is chosen, [Storybook mandatory configurations](#storybook-mandatory-configurations) will be automatically added.
+
+### Storybook mandatory configurations
+
+Storybook will require 2 files in the folder `.storybook` added into the root folder of the project.
+
+- **main.js**: partial webpack configuration used by Storybook to build the stories. It is used by Otter to add custom CSS (including theming variables)
+- **preview.js**: set of functions and default configurations used by the Storybook before displaying a component preview. This is used by Otter to load metadata (configuration, localization and styling) to setup the component controllers and integrate localization into the display.
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:31]: set up

File: packages/@o3r/storybook/README.md [hunk 52]
```diff
@@ -1,3 +1,112 @@
...
+
+To generate a basic story when generating a new component, the configuration `useStorybook` will be added into the *angular.json* file.
+
+## Component generator
+
+The Otter component generator (from `@o3r/core`) will generate the `<component>.stories.ts` file if the parameter `useStorybook` from the *angular.json* is detected. The parameter can be also overridden by the `--useStorybook` option of the generator.
+
+The generated story will come with the following supported feature:
+
+- **Localization**: default translation will be display on the previewed component.
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:43]: displayed

File: packages/@o3r/storybook/README.md [hunk 53]
```diff
@@ -1,3 +1,112 @@
...
+
+## Otter AddOns
+
+To facilitate the customization of an Otter component, 2 Storybook Addons are provided.
+
+### Global Theming tab
+
+- ***(1)***: To facilitate the administration of global theming of a component, a tab has been added to the default storybooks tabs.
+- ***(2)***: Each global theme variables are editable
+- ***(3)***: 3 actions buttons are provided
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:57]: action buttons

File: packages/@o3r/storybook/README.md (hunk 27 above)
Comment (sdo-1A) [reply packages/@o3r/storybook/README.md:58]: multi theme **configuration**

File: packages/@o3r/storybook/README.md [hunk 54]
```diff
@@ -1,3 +1,112 @@
...
+3 different exports are available:
+
+- **Export Theme**: exports global theme as CSS file.
+- **Export Configuration**: exports component configuration as JSON file.
+- **Export Component style**: exports component variable override as CSS file.
+
+## Stories Helpers
+
+Otter provides 3 helpers to generate automatically the component controllers.
+Each helpers should be used in both `preview.js` to initialize the value and in `<component>.stories.ts` file to retrieve the value.
```
Comment (sdo-1A) [inline packages/@o3r/storybook/README.md:73]: helper

File: packages/@o3r/dev-tools/README.md [hunk 55]
```diff
@@ -1,3 +1,283 @@
...
+```
+
+## PR Artifact cleaner
+
+Clean old PR artifacts by identifying using the the build version that is present in the path. If build version is not present in the path this tool cannot yet be used.
+
+### Usage
+
+```bash
+Usage: pr-artifact-cleaner -b <Base 64 encoding of username:password (password already encrypted from artifactory UI)> [options]
```
[the remark this answers is not in the document]
Comment (kpanot) [reply packages/@o3r/dev-tools/README.md:155]: This is the --help output, we can fix it in another PR/Issue

**Review by cpaulve-1A — APPROVED** [review]

**Review by sdo-1A — APPROVED** [review]

**Pull request merged**